connexion/decorators/response.py

```diff
@@ -69,8 +71,10 @@ def wrapper(*args, **kwargs):
             try:
                 data, status_code, headers = self.get_full_response(result)
                 self.validate_response(data, status_code, headers)
-            except NonConformingResponse as e:
-                return problem(500, 'Internal Server Error', e.reason)
+            except NonConformingResponseBody as e:
+                return problem(500, 'Response body does not conform to specification', str(e))
```

Comment:
If you move those default error messages to their respective classes (`NonConformingResponseBody` and `NonConformingResponseHeaders`) this code could be simplified to be only:
```python
except NonConformingResponse as e:
    return problem(500, e.message, str(e))
```